lib/windows/app/components/DeviceSelector.jsx

```diff
@@ -35,295 +35,124 @@
-                            className={dropdownCssClass}
-                            title={selectorText}
-                        />
-                        <DropdownMenu
-                            id="device-selector-list"
-                            className={dropdownMenuCssClass}
-                        >
-                            { this.renderDeviceItems() }
-                            { this.renderCloseItem() }
-                        </DropdownMenu>
-                    </Dropdown>
-                </div>
-            </span>
+            <Dropdown id="device-selector" className={cssClass} disabled={devices.length === 0}>
+                <DropdownToggle
+                    className={dropdownCssClass}
+                    title={togglerText}
+                />
+                <DropdownMenu
+                    id="device-selector-list"
+                    className={dropdownMenuCssClass}
+                >
+                    { devices.map(device => this.getItemFromDevice(device)) }
+                    { closeItem }
+                </DropdownMenu>
+            </Dropdown>
         );
     }
 }
 
 DeviceSelector.propTypes = {
-    devices: PropTypes.oneOfType([
-        PropTypes.instanceOf(Array),
-        PropTypes.instanceOf(Iterable),
-    ]).isRequired,
-    selectedDevice: PropTypes.shape({
-        serialNumber: PropTypes.serialNumber,
-    }),
-    isLoading: PropTypes.bool,
-    isExpanded: PropTypes.bool,
-    isSerialPortVerificationEnabled: PropTypes.bool,
-    toggleExpanded: PropTypes.func.isRequired,
     onSelect: PropTypes.func.isRequired,
     onDeselect: PropTypes.func.isRequired,
-    bindHotkey: PropTypes.func.isRequired,
-    hotkeyExpand: PropTypes.string,
     cssClass: PropTypes.string,
     dropdownCssClass: PropTypes.string,
     dropdownMenuCssClass: PropTypes.string,
     menuItemCssClass: PropTypes.string,
     menuItemDetailsCssClass: PropTypes.string,
-    filter: PropTypes.func,
-    compareFunction: PropTypes.func,
-    deviceNameParser: PropTypes.func,
-    deviceDetailsParser: PropTypes.func,
-    deviceKeyParser: PropTypes.func,
+    togglerText: PropTypes.string.isRequired,
+    displayCloseItem: PropTypes.bool.isRequired,
+    devices: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
 };
 
 DeviceSelector.defaultProps = {
-    selectedDevice: null,
-    isExpanded: false,
-    isLoading: false,
-    isSerialPortVerificationEnabled: true,
-    indicatorStatus: 'off',
-    hotkeyExpand: 'Alt+P',
     cssClass: 'core-padded-row',
     dropdownCssClass: 'core-device-selector core-btn btn-primary',
     dropdownMenuCssClass: 'core-dropdown-menu',
     menuItemCssClass: 'core-device-selector-item btn-primary',
     menuItemDetailsCssClass: 'core-device-selector-item-details',
-    filter: deviceFilter,
-    compareFunction: deviceCompareFunction,
-    deviceNameParser: getDeviceDisplayName,
-    deviceDetailsParser: getDeviceDisplayDetails,
-    deviceKeyParser: getDeviceKey,
```

Comment:
I guess it is ok to remove these props. Allowing apps to add custom filtering, sorting, etc. is probably feature creep. Just have to be aware that the apps have to replace the complete selector component to make such adjustments from now on.